src/components/crudList.js

```diff
@@ -0,0 +1,38 @@
+import {
+  data, addData, saveData, removeCompleted,
+} from './statusUpdate';
+import { clearData } from './interaction';
+
+const createTask = (e) => {
+  const lastItem = data[data.length - 1];
+  const newTask = {
+    description: e.target.value,
+    completed: false,
+    index: lastItem ? lastItem.index + 1 : 0,
+  };
+  addData(newTask);
+};
+
+const updateTask = (index, elm) => {
+  const findObj = data.find((obj) => obj.index === index);
+  findObj.description = elm.value;
+  saveData();
+};
+
+const clearAllCompleted = () => {
+  const completed = data.filter((obj) => !obj.completed);
+  clearData();
+  removeCompleted(completed);
+  saveData();
+};
+
+const deleteTask = (index) => {
+  const removed = data.filter((obj) => obj.index !== index);
+  clearData();
+  removeCompleted(removed);
+  saveData();
+};
```

Comment:
- [ ] Please add a method to update all the indexes of the remaining elements every time you delete one or more elements from the array.

Reply:
:heavy_check_mark: Well Done